docs/notes/2.21.x.md

```diff
@@ -0,0 +1,153 @@
+# 2.21.x Release Series
+
+Pants 2 is a fast, scalable, user-friendly build system for codebases of all sizes. It's currently focused on Python, Go, Java, Scala, Kotlin, Shell, and Docker, with support for other languages and frameworks coming soon.
+
+Individuals and companies can now [sponsor Pants financially](https://www.pantsbuild.org/sponsorship).
+
+Pants is an open-source project that is not owned or controlled by any one company or organization, and does incur some expenses. These expenses are managed by Pants Build, a non-profit that was established for this purpose. This non-profit's only source of revenue is sponsorship by individuals and companies that use Pants.
+
+We offer [formal sponsorship tiers for companies](https://www.pantsbuild.org/sponsorship), as well as individual sponsorships via [GitHub](https://github.com/sponsors/pantsbuild).
+
+## What's New
+
+### Highlights
+
+- All built-in backends now support the `[test].attempts_default` option to automatically retry tests.
+- Significant speed-ups when manipulating Python dependencies, especially huge ones like PyTorch.
+- When a `pants test ...` run in CI fails, the summary now includes suggested invocation to copy-paste and re-run locally.
+- The `__defaults__` symbol now applies to targets generated by a target generator, in addition to targets written literally in BUILD files.
```

Comment:
I've just tried `2.21.0a0` and it slightly broke our expectations. We had 
```python
__defaults__(
    extend=True,
    all=dict(
        # This will set PYTHON_IMAGE build arg in all docker images by default.
        extra_build_args=["PYTHON_IMAGE=python:3.11"],
        # This will tell pants that all python sources need to use python 3.11 by default.
        interpreter_constraints=["==3.11.*"],
    ),
)
```
now we have to do
```python
__defaults__(
    {
        (docker_image,): dict(
            # This will set PYTHON_IMAGE build arg in all docker images by default.
            extra_build_args=["PYTHON_IMAGE=python:3.11"],
        )
    },
    all=dict(
        # This will tell pants that all python sources need to use python 3.11 by default.
        interpreter_constraints=["==3.11.*"],
    ),
    extend=True,
)
```
This happened because `pex_binary.extra_build_args` was added, and it has the same name as `docker_image.extra_build_args`. Maybe it's good to add a note about it

Reply:
> I've just tried 2.21.0a0 and it slightly broke our expectations

Thanks for trying! Did you find some other issues or was it otherwise okay?

> This happened because pex_binary.extra_build_args was added, and it has the same name as docker_image.extra_build_args. Maybe it's good to add a note about it

I think that'll be an issue with the addition of `extra_build_args` to `pex_binary` that you've identified, rather than with this `__defaults__` change. I guess it's unlucky they have the same name but different meanings... but this is somewhat fundamental to `__defaults__(all=...)`. I've filed https://github.com/pantsbuild/pants/issues/20925.

I'll add a note to the release notes too.

Reply:
Thanks for finding this @grihabor!

Reply:
> Did you find some other issues or was it otherwise okay?

No, it's the only thing

> I'll add a note to the release notes too.

👍